packages/engine/vscode/src/lib/filemanager.ts

```diff
@@ -0,0 +1,150 @@
+import { Folder } from "./types";
+import { CommandPlugin } from "@remixproject/engine-vscode";
+import { window, workspace, Uri, FileType, commands, FileStat } from "vscode";
+
+const profile = {
+  name: "fileManager",
+  displayName: "Native Filemanager for Remix vscode plugin",
+  description: "Provides communication between vscode filemanager and remix-plugin",
+  kind: "filemanager",
+  permission: true,
+  location: "sidePanel",
+  documentation: "https://remix-ide.readthedocs.io/en/latest/solidity_editor.html",
+  version: "0.0.1",
+  methods: [
+    "getFolder",
+    "getCurrentFile",
+    "getFile",
+    "setFile",
+    "switchFile",
+    // NextFileSystemAPI
+    "open",
+    "writeFile",
+    "readFile",
+    "rename",
+    "copyFile",
+    "mkdir",
+    "readdir",
+  ],
+};
+
+/**
+ * TODO:
+ * it should be possible to install https://github.com/ethereum/remix-plugins-directory and
+ * import `NextFileSystemApi` interface and implement `FileManagerPlugin` following that interface
+ * `export default class FileManagerPlugin extends CommandPlugin implements NextFileSystemApi`
+ */
+
+export default class FileManagerPlugin extends CommandPlugin {
+  constructor() {
+    super(profile);
+  }
+  /** Open the content of the file in the context (eg: Editor) */
+  open(path: string): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      let uri = Uri.file(path);
+      commands.executeCommand('vscode.open', uri)
+        .then(() => {
+          resolve(true);
+        })
+    })
+  }
+  /** Set the content of a specific file */
+  writeFile(path: string, data: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+      const turi = Uri.parse(path);
+      const encoder = new TextEncoder();
```

Comment:
Where does the TextEncoder comes from ?

Reply:
From typescript.

ref: https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder